### PR TITLE
feat: client changes for query stats 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

this PR adds the query statistics for the logs explorer page 

#### Related Issues / PR's

fixes https://github.com/SigNoz/engineering-pod/issues/1661

#### Screenshots

https://github.com/user-attachments/assets/77b86769-77cd-4dde-94e9-12b18150acd8

### CODE REVIEW

**Comment by vikrantgupta25** [comment]
Do not merge this, raised against the feature branch

File: frontend/src/api/common/getQueryStats.ts
Comment (vikrantgupta25) [inline frontend/src/api/common/getQueryStats.ts:19]: ignore this for now, will change this

File: frontend/src/api/common/getQueryStats.ts
Comment (SagarRajput-7) [inline frontend/src/api/common/getQueryStats.ts:31]: any safety check needed here? specially for `event: Event` type, that imo dont by default have data prop

File: frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss
Comment (SagarRajput-7) [inline frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss:10]: empty styles

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss:4]: removed

File: frontend/src/api/common/getQueryStats.ts
Comment (vikrantgupta25) [reply frontend/src/api/common/getQueryStats.ts:31]: added

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:124]: Can we not use a fixed constant here ?

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:340]: use uuid library

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:352]: Reason for delay ?

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:712]: Please make sure fallbacks are in place here. Bad data shouldn't break the UI

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx:124]: no we cannot use a fixed constant as multiple users opening the UI at the same time will have the same id and the socket will get attached to wrong queries if we have the same initial ID for everyone

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx:352]: there is a delay between the `query_range` API reaching the query service and ws connection, if the connection is being tried to establish before the query_range API reaching the qs it fails hence the delay.

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx:712]: if we do not receive any data we do not show anything in the UI, just the status of the request.

File: frontend/src/container/LogsExplorerViews/index.tsx
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx]: done

**Review by YounixM — DISMISSED** [review]
LGTM. 

Please get review from a BE team member for API related changes.

**Comment by vikrantgupta25** [comment]
bad rebase! reopened here - https://github.com/SigNoz/signoz/pull/5706

**Pull request closed without merging**
